packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx

```diff
@@ -85,8 +85,10 @@ const AdaptedVirtualSelect = reduxFormFieldAdapter({
 });
 
 const serviceFormSelector = formValueSelector('serviceForm');
-const oneHourInMilliSeconds = 3600000;
 const oneMinuteInMilliSeconds = 60000;
+const oneHourInMilliSeconds = 3600000;
+const stepDivider = 240;
```

Comment:
honestly i tried to find a num so we can get good accureancy, good performance and that all time ranges are divisible by it.
what exacly do you want me to write? :)

Reply:
Did you consider my comment on the issue about the desired number of pixels for the plot?

Reply:
I'm not sure this is a good direction.
I thought maybe to do what kibana does:
https://github.com/elastic/kibana/blob/main/src/plugins/data/common/search/aggs/buckets/lib/histogram_calculate_interval.ts
i thought that our use case is much simpler so I did in this way

Reply:
👋🏻 Hi,
I think 240 is a good number.
I will save a lot of time.

But think of it like this:
The larger graph is 600px width on my monitor - it makes some sense to have 240 points on the graph
The smaller graph is 100px fixed - it doesn't make sense to have 240 points on the graph.

What do you think about adding a modifier to consider the target Viz size?

Reply:
different amount of data points sounds good to me too.

Reply:
@AlonMiz WDYT?

Reply:
@yurishkuro WDYT?

Reply:
I think we all want different amount of data points, the question is how many.

> The larger graph is 600px width on my monitor - it makes some sense to have 240 points on the graph

I disagree with that. It means that even if you have a high resolution data source that can provide more than 600 data points, you're forcing it to limit to only 240 data points, which can only be achieved with additional aggregation that loses the details like small spikes, even though the 600px graph is capable of showing more details. This completely artificial & unnecessary degradation of results doesn't sit well with me. What if I have high-res wide screen and my graph can stretch to 2000px? 240 data points would be like every 8th pixel, with fake straight interpolation line between then.

Reply:
+1 to snapping intervals to whole ticks (I would start from 1s, 10s, 30s, 1m...)

fwiw, the google chart is 1m buckets over 7hrs ==> 420 points across about 1000px (on my screen)

Reply:
IMO, it doesn't matter if we have buckets that (make sense) I would argue that labels below the graph matter, and they are not related to the actual points of the graph other than to orientate the graph

Reply:
some code we had long time ago, still relevant

```
const intervalForBigChart = calculateInterval(from, to, 600);
const intervalForSmallChart = calculateInterval(from, to, 50);
```

```
import _ from 'lodash';

const time = new (function () {
  this.second = 1000;
  this.minute = 60 * this.second;
  this.hour = 60 * this.minute;
  this.day = 24 * this.hour;
  this.week = 7 * this.day;
  this.year = 365 * this.day;
})();

const intervalMin = [
  ['100ms', time.second * 0.5, time.second * 0.1], // < 0.5s
  ['1s', time.second * 5, time.second * 1], // < 5s
  ['5s', time.second * 7.5, time.second * 5], // < 7.5s
  ['10s', time.second * 15, time.second * 10], // < 15s
  ['30s', time.second * 45, time.second * 30], // < 45s
  ['1m', time.minute * 3, time.minute * 1], // < 3m
  ['5m', time.minute * 9, time.minute * 5], // < 9m
  ['10m', time.minute * 20, time.minute * 10], // < 20m
  ['30m', time.minute * 45, time.minute * 30], // < 45m
  ['1h', time.hour * 2, time.hour * 1], // < 2h
  ['3h', time.hour * 6, time.hour * 3], // < 6h
  ['12h', time.hour * 24, time.hour * 12], // < 24h
  ['24h', time.day * 2, time.day * 1], // < 2d
  ['2d', time.week, time.day * 2], // < 1w
  ['1w', time.week * 3, time.week * 1], // < 3w
  ['30d', time.week * 6, time.day * 30], // < 6w
  ['60d', time.year * 2, time.day * 60], // < 2y
  ['1y', Infinity], // OMG
];

const roundInterval = (interval: number) => {
  let chosen = ['24h', time.day * 2, time.day * 1];

  _.forEach(intervalMin, limit => {
    if (interval < limit[1]) {
      chosen = limit;

      return false; // this will break lodash forEach
    }
  });

  return chosen;
};

const calculateInterval = (from: number, to: number, size = 200) => roundInterval((to - from) / size)[0];

const getIntervalSec = (from: number, to: number, size = 200) => roundInterval((to - from) / size)[2];

export const intervalCalculatorService = {
  calculateInterval,
  getIntervalSec,
};

```